### ISSUE TITLE
Explicit DocumentationFile=path instead of GenerateDocumentationFile=true leads to build race

### ISSUE THREAD

**Issue by nguerrera** [dotnet/sdk#1599]
Previously, C# projects would set an explicit intermediate path for the documentation file. This leads to using the same intermediate path for different TFMs that are built in parallel. We implemented GenerateDocumentationFile=true, but users porting existing projects might not know about it. 

While the fix is simple (use GenerateDocumentationFile), the error is intermittent and doesn't at all hint at the fix: "cannot write to obj\Debug\Something.xml: already in use".

We should detect when  the DocumentationFile is going to be written to the intermediate output path minus TFM in a multi-targeted project and issue an error. Alternatively, we could implement a fixup to insert the TFM in to the DocumentationFile path.

See also:
* https://github.com/dotnet/roslyn/pull/22232 
* https://github.com/dotnet/sdk/issues/1598

**Comment by vlada-shubina** [comment]
> @vlada-shubina, did you close this by mistake? [d8bb6c2](https://github.com/dotnet/sdk/commit/d8bb6c240d10ae4678bb7cd9f23616b10ef136a1) seems to relate to [dotnet/templating#1599](https://github.com/dotnet/templating/issues/1599) and not to this #1599.

yes, it was incorrectly closed. I will check what else might be affected shortly
